src/tests/unit/utils.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { expect } from "chai";
+import { AppStoreState } from "../../app/stores";
+import { deepClone } from "../../core";
+import { DefaultAttributes } from "../../core/prototypes";
+
+export function makeDefaultAttributes(state: AppStoreState) {
+  const defaultAttributes: DefaultAttributes = {};
+  const { elements } = state.chart;
+  for (let i = 0; i < elements.length; i++) {
+    const el = elements[i];
+    defaultAttributes[el._id] = deepClone(
+      state.chartState.elements[i].attributes
+    );
+  }
+  return defaultAttributes;
+}
+
+/** Test if a deep equals b with tolerance on numeric values */
+export function expect_deep_approximately_equals(
+  a: any,
+  b: any,
+  tol: number,
+  context?: any
+) {
+  try {
+    if (a == null && b == null) {
+      return;
+    }
+    if (a == null || b == null) {
```

Comment:
Consider adding check if both a & b are not the same `typeof` here.

Reply:
In my opinion, it will be superfluous. This function will work if a, b is null/undefine. For other cases there are `else if` and `else`